server/text_generation_server/models/custom_modeling/mpt_modeling.py

```diff
@@ -983,6 +1027,11 @@ def forward(
                 )
             pos_emb = self.wpe(pos)
             x = tok_emb + pos_emb
+            if self.embedding_fraction != 1:
+                x = (
+                    x * self.embedding_fraction
+                    + x.detach() * (1 - self.embedding_fraction)
+                )
```

Comment:
I'm not sure I understand the purpose of this.

detaching, has no effect at inference time, has it ? If it has no effect, then this is essentially a no-op, and we could safely ignore embedding_fraction, no ? Am I missing something ?

Reply:
You are right, is back to x after the computation so this part of the code is actually redundant. Thanks for pointing out.